### PR TITLE
<fix>[block]:<class component>

### PR DESCRIPTION
**Pull request by niomu**
/attempt #488

### CODE REVIEW

File: packages/million/block.ts [hunk 1]
```diff
@@ -46,7 +46,16 @@ export const block = (
...
+  const vnode =
+    typeof fn === 'function' && !!fn.prototype
+      ? new (fn as any)(HOLE_PROXY).render()
+      : typeof fn === 'object'
+      ? new Proxy(fn, {
+          get(_, key: string): Hole {
+            return { $: key };
+          },
+        })
+      : fn(HOLE_PROXY);
```
Comment (AbhiPrasad) [inline packages/million/block.ts:58]: I don't understand how this allows for blocks to become cross file - could you add some context in your PR description about how this approach works?

This change doesn't make any changes to the compiler, so we'll still run into issues there.

Comment (niomu) [reply packages/million/block.ts:58]: This is just to support class component for now

File: package.json [hunk 2]
```diff
@@ -106,9 +106,12 @@
     "unplugin": "^1.3.1"
   },
   "devDependencies": {
-    "@types/babel-types": "^7.0.11",
+    "@emotion/react": "^11.11.1",
+    "@emotion/styled": "^11.11.0",
+    "@mui/material": "^5.14.5",
```
[the remark this answers is not in the document]
Comment (niomu) [reply package.json:111]: Yes it is for the test

**Comment by niomu** [comment]
Still working on it... Just opened the PR to let you know

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (AbhiPrasad) [reply package.json:111]: Ah I meant we should avoid adding in new dependencies just for tests, could we remove `@mui/material`?

[the remark this answers is not in the document]
Comment (niomu) [reply package.json:111]: Yeah no problem, I will

**Pull request closed without merging**